asapdiscovery-docking/asapdiscovery/docking/workflows/large_scale_docking.py

```diff
@@ -298,26 +310,40 @@ def large_scale_docking(inputs: LargeScaleDockingInputs):
         molfile = MolFileFactory.from_file(inputs.filename)
         query_ligands = molfile.ligands
 
-    # load complexes from a directory or from fragalysis
+    # load complexes from a directory, from fragalysis or from a pdb file
     if inputs.structure_dir:
         logger.info(f"Loading structures from directory: {inputs.structure_dir}")
         structure_factory = StructureDirFactory.from_dir(inputs.structure_dir)
         complexes = structure_factory.load(
             use_dask=inputs.use_dask, dask_client=dask_client
         )
-    else:
+    elif inputs.fragalysis_dir:
         logger.info(f"Loading structures from fragalysis: {inputs.fragalysis_dir}")
         fragalysis = FragalysisFactory.from_dir(inputs.fragalysis_dir)
         complexes = fragalysis.load(use_dask=inputs.use_dask, dask_client=dask_client)
 
+    elif inputs.pdb_file:
+        logger.info(f"Loading structures from pdb: {inputs.pdb_file}")
+        complex = Complex.from_pdb(
+            inputs.pdb_file,
+            target_kwargs={"target_name": inputs.pdb_file.stem},
+            ligand_kwargs={"compound_name": f"{inputs.pdb_file.stem}_ligand"},
+        )
+        complexes = [complex]
+
+    else:
+        raise ValueError(
+            "Must specify either fragalysis_dir, structure_dir or pdb_file"
+        )
```

Comment:
isn't this logic duplicated from 205? I guess git rid of that section, having it here makes sense

Reply:
Yeah I just hate un-terminated if-else that doesn't make the fallthrough case explicit.